python/cudf/setup.py

```diff
@@ -77,7 +77,7 @@
     ],
     # Include the separately-compiled shared library
     setup_requires=["cython"],
-    ext_modules=cythonize(extensions, nthreads=nthreads),
+    ext_modules=cythonize(extensions, nthreads=nthreads, language_level="3"),
```

Comment:
Can we move all of the compiler directives to here? I.e. `profile=False`, `language=c++`, `embedsignature=True`, `boundscheck=False`? If so we can get rid of the header from all of our Cython files and not worry about it moving forward 😄

Reply:
Yes we can, but be careful to only include ones that should apply globally. I'm not sure if `boundscheck=False` should apply everywhere or not, since the warning only showed up in the `io` modules.

Reply:
Okay, well how about lets do it for everything other than `boundscheck` for now?

Reply:
Yep, that sounds like a good idea.

Reply:
![image](https://user-images.githubusercontent.com/178183/75080742-baf10380-54c1-11ea-8e45-0ec52fbdddd3.png)

Do the recent commits resolve this, or are there other compiler directives we should include?